test/github_issue_802.cpp

```diff
@@ -0,0 +1,28 @@
+// Copyright 2025 Matt Borland
+// Distributed under the Boost Software License, Version 1.0.
+// https://www.boost.org/LICENSE_1_0.txt
+
+#include <boost/decimal.hpp>
+#include <boost/core/lightweight_test.hpp>
+#include <iostream>
+
+void fasting()
```

Comment:
Not sure adding this as test is useful. 
Result was always correct afaik.
This is hard to test but if you want one way to test this is to introduce detail:: helper that checks if 2 numbers can be multiplied within range of u128, and then test that helper. 
Might be an overkill, but it is basically just extracting
```
    const auto lhs_dig {detail::num_digits(lhs_sig)};
    const auto rhs_dig {detail::num_digits(rhs_sig)};
    return lhs_dig + rhs_dig <= std::numeric_limits<uint128>::digits10;
    
```


into bool returning helper function.

Reply:
It's a good reproducer for using the debugger. I'm not super worried about the test in this case since the special functions test suites will for sure hit both paths.